compiler-core/src/build/package_loader.rs

```diff
@@ -112,10 +110,13 @@ where
         // which should be loaded.
         let mut inputs = self.read_sources_and_caches()?;
 
-        // Check for any removed modules, by looking at cache files that don't exist in inputs
-        for cache_file in gleam_cache_files(&self.io, &self.artefact_directory) {
-            let module = module_name(&self.artefact_directory, &cache_file);
+        // Check for any removed modules, by looking at cache files that don't exist in inputs.
+        // Delete the cache files for removed modules and mark them as stale
+        // to trigger refreshing dependent modules.
+        for module in CacheFiles::modules_with_meta_files(&self.io, &self.artefact_directory) {
             if (!inputs.contains_key(&module)) {
+                tracing::debug!(%module, "module_removed");
+                CacheFiles::new(&self.artefact_directory, &module).delete(&self.io);
```

Comment:
This line is the fix for the temporarily deleted files not being always recompiled.